OpenLIFUHome/Resources/UI/OpenLIFUHome.ui

```diff
@@ -186,6 +186,13 @@
         </property>
        </widget>
       </item>
+      <item row="1" column="0">
+       <widget class="QPushButton" name="installPythonReqsButton">
```

Comment:
Did you intend to put this button inside of the "Advanced" tab.  I think it should be up near the top, since in the extension version, you will need to click this in order to use the modules.

One thing I have done in another app is to check for the installed packages when entering the module, and then use that result to show or set relevant text on the install button, and after the install the button is hidden or set with different text (re-install vs install)

To make the custom app look nice i think we want show/hide, with an additional re-install button in the advanced section.

Reply:
Yes I meant to put it inside "Advanced". 

I didn't want to check for installed packages when entering the module, because sometimes the import of openlifu takes a moment (https://github.com/OpenwaterHealth/OpenLIFU-python/issues/8) 

I was imagining that whenever the user does something that requires the openlifu package, that would trigger the check and install function to get called. So that's why I am hiding away the button under "Advanced" -- because it wouldn't normally be needed.

I like the idea of changing the button text to "re-install" appropriately, but again I didn't want to attempt `import openlifu` unnecessarily when it can cause a moment of lag

What do you think?

I could change the way in which the check is done. Instead of checking by attempting `import openlifu`, I could check using pip itself, something like `slicer.util._executePythonModule("pip", ["show", "openlifu"])` ... IDK if there is a better way to check if something is installed

Reply:
Yes I meant to put it inside "Advanced". 

I didn't want to check for installed packages when entering the module, because sometimes the import of openlifu takes a moment (https://github.com/OpenwaterHealth/OpenLIFU-python/issues/8) 

I was imagining that whenever the user does something that requires the openlifu package, that would trigger the check and install function to get called. So that's why I am hiding away the button under "Advanced" -- because it wouldn't normally be needed.

I like the idea of changing the button text to "re-install" appropriately, but again I didn't want to attempt `import openlifu` unnecessarily when it can cause a moment of lag

What do you think?

I could change the way in which the check is done. Instead of checking by attempting `import openlifu`, I could check using pip itself, something like `slicer.util._executePythonModule("pip", ["show", "openlifu"])` ... IDK if there is a better way to check if something is installed

Reply:
If you are planning to trigger the check and install through other actions, that fine and we can leave the button in Advanced.

For other methods to check if the module exists, maybe https://docs.python.org/3/library/importlib.html#importlib.util.find_spec  (from https://stackoverflow.com/questions/14050281/how-to-check-if-a-python-module-exists-without-importing-it)

Reply:
Nice, this will be a better way. I will do this and request review again when it is ready